slides/code-formatting/index.html

```diff
@@ -0,0 +1,395 @@
+    end
+            </code></pre>
+          </section>
+
+          <section>
+            <div>On line 3 there is no keyword. So neither of the rules apply and we leave it as is:</div>  
+
+            <pre><code class="hljs ruby" data-line-numbers>def widths(row)
+  row.map do |cell|
+    cell.width
+    end
+    end
+            </code></pre>
+          </section>
+
+          <section>
+            <div>Line 4 has an <pre class="pre-inline">end</pre>. Here we need to use <b>Rule 2</b>, which says that it gets indented one level less deep (-1 level) than the previous line:</div>  
+
+            <pre><code class="hljs ruby" data-line-numbers>def widths(row)
+  row.map do |cell|
+    cell.width
+  end
+    end
+            </code></pre>
+          </section>
+
+          <section>
+            <div>On line 5 the same rule applies again: We indent it one level less deep (-1 level) than the previous line:</div>  
+
+            <pre><code class="hljs ruby" data-line-numbers>def widths(row)
+  row.map do |cell|
+    cell.width
+  end
+end
+            </code></pre>
+
+            <p>And we're done!</p>
+          </section>
+        </section>
+
+        <section>
+          <h2>Interactive exercise</h2>
+
+          <p>Let's exercise the rules together with another example:</p>
+
+          <pre class="fragment fade-in"><code class="hljs ruby">def judge_weather(weather)
+if weather == "sunny"
+puts "The weather is great!"
+3.times do
+puts "Yes!"
+end
+elsif weather == "rainy"
+puts "Better stay inside..."
+end
+end</code></pre>
+        </section>
+
+        <section>
+          <section>
+            <h2>Pattern recognition</h2>
+
+            <p class="fragment fade-in">If done correctly, indentation lets you see immediately which lines of code belong together and which don't:</p>
+
+            <pre class="fragment fade-in"><code class="hljs ruby" data-line-numbers>def judge_weather(weather)
+  if weather == "sunny"
+    puts "The weather is great!"
+    3.times do
+      puts "Yes!"
+    end
+  elsif weather == "rainy"
+    puts "Better stay inside..."
+  end
+end
+            </code></pre>
+
+            <div class="fragment fade-in">Here you can see easily that lines 3-6 are executed only if the <pre class="pre-inline">weather</pre> is "sunny".</div>
+          </section>
+
+          <section>
+            <div class="fragment fade-in">It's not uncommon to have many nested levels. Sometimes this results in a lot of <pre class="pre-inline">end</pre>s at the bottom of your file:</p>
+
+            <pre class="fragment fade-in"><code class="hljs ruby" data-line-numbers>def ...
+  if ...
+    do
+      if ...
+        if ...
+          ...
+        end
+      end
+    end
+  end
+end
+            </code></pre>
+          </section>
+        </section>
+
+        <section>
+          <section>
+            <h2>Hints</h2>
+
+            <p class="fragment fade-in">
+              Never skip indentation levels from one line to the other:
```

Comment:
```suggestion
              Avoid skipping indentation levels from one line to the other:
```

_Never_ is such as strong word ... 😄

Reply:
Alright, I guess... ;P